sdk/python/kfp_tekton/compiler/compiler.py

```diff
@@ -424,12 +424,18 @@ def _create_pipeline_workflow(self, args, pipeline, op_transformers=None, pipeli
             'taskSpec': template['spec'],
           }
 
-        if template['metadata'].get('labels', None):
-          task_ref['taskSpec']['metadata'] = task_ref['taskSpec'].get('metadata', {})
-          task_ref['taskSpec']['metadata']['labels'] = template['metadata']['labels']
-        if template['metadata'].get('annotations', None):
-          task_ref['taskSpec']['metadata'] = task_ref['taskSpec'].get('metadata', {})
-          task_ref['taskSpec']['metadata']['annotations'] = template['metadata']['annotations']
+        task_metadata = task_ref['taskSpec'].get('metadata', {})
+        task_ref['taskSpec']['metadata'] = task_metadata
+        task_metadata['labels'] = template['metadata'].get('labels', {})
+        task_metadata['labels']['pipelines.kubeflow.org/pipelinename'] = task_metadata['labels'].get('pipelines.kubeflow.org/pipelinename', '')
```

Comment:
Not sure will the pipeline level labels will overwrite the task one, but make sure to check the pipeline level labels in this logic as well

Reply:
Pipeline level labels are injected over here https://github.com/kubeflow/kfp-tekton/blob/master/sdk/python/kfp_tekton/compiler/compiler.py#L568-L574